e3sm_diags/derivations/utils.py

```diff
@@ -25,70 +25,77 @@ def aplusb(var1: xr.DataArray, var2: xr.DataArray, target_units=None):
 
 
 def convert_units(var: xr.DataArray, target_units: str):  # noqa: C901
-    if var.attrs.get("units") is None:
-        if var.name == "SST":
-            var.attrs["units"] = target_units
-        elif var.name == "ICEFRAC":
-            var.attrs["units"] = target_units
-            var = 100.0 * var
-        elif var.name == "AODVIS":
-            var.attrs["units"] = target_units
-        elif var.name == "AODDUST":
-            var.attrs["units"] = target_units
-    elif var.name == "FAREA_BURNED":
-        var = var * 1e9
-        var.attrs["units"] = target_units
-    elif var.attrs["units"] == "gC/m^2":
-        var = var / 1000.0
-        var.attrs["units"] = target_units
-    elif var.name == "FLOODPLAIN_VOLUME" and target_units == "km3":
-        var = var / 1.0e9
-        var.attrs["units"] = target_units
-    elif var.name == "AOD_550_ann":
-        var.attrs["units"] = target_units
-    elif var.name == "AOD_550":
-        var.attrs["units"] = target_units
-    elif var.attrs["units"] == "C" and target_units == "DegC":
-        var.attrs["units"] = target_units
-    elif var.attrs["units"] == "N/m2" and target_units == "N/m^2":
-        var.attrs["units"] = target_units
-    elif var.name == "AODVIS" or var.name == "AOD_550_ann" or var.name == "TOTEXTTAU":
-        var.attrs["units"] = target_units
-    elif var.attrs["units"] == "fraction":
-        var = 100.0 * var
-        var.attrs["units"] = target_units
-    elif var.attrs["units"] == "mb":
-        var.attrs["units"] = target_units
-    elif var.attrs["units"] == "gpm":  # geopotential meter
-        var = var / 9.8 / 100  # convert to hecto meter
-        var.attrs["units"] = target_units
-    elif var.attrs["units"] == "Pa/s":
-        var = var / 100.0 * 24 * 3600
-        var.attrs["units"] = target_units
-    elif var.attrs["units"] == "mb/day":
-        var = var
-        var.attrs["units"] = target_units
-    elif var.name == "prw" and var.attrs["units"] == "cm":
-        var = var * 10.0  # convert from 'cm' to 'kg/m2' or 'mm'
-        var.attrs["units"] = target_units
-    elif var.attrs["units"] in ["gC/m^2/s"] and target_units == "g*/m^2/day":
-        var = var * 24 * 3600
-        var.attrs["units"] = var.attrs["units"][0:7] + "day"
-    elif (
-        var.attrs["units"] in ["gN/m^2/s", "gP/m^2/s"] and target_units == "mg*/m^2/day"
-    ):
-        var = var * 24 * 3600 * 1000.0
-        var.attrs["units"] = "m" + var.attrs["units"][0:7] + "day"
-    elif var.attrs["units"] in ["gN/m^2/day", "gP/m^2/day", "gC/m^2/day"]:
-        pass
-    else:
-        original_udunit = cf_units.Unit(var.attrs["units"])
-        target_udunit = cf_units.Unit(target_units)
-
-        var.values = original_udunit.convert(var.values, target_udunit)
-        var.attrs["units"] = target_units
-
-    return var
+    var_new = var.copy()
+
+    with xr.set_options(keep_attrs=True):
```

Comment:
`convert_units()` was missing `xr.set_options(keep_attrs=True)`, which lead to any Xarray arithmetic to drop attributes including `units`.